src/objectExplorer/objectExplorerProvider.ts

```diff
@@ -70,68 +70,53 @@ export class ObjectExplorerProvider implements vscode.TreeDataProvider<any> {
         return this._objectExplorerService.expandNode(node, sessionId, promise);
     }
 
-    public getConnectionCredentials(sessionId: string): IConnectionInfo {
-        if (sessionId) {
-            return this._objectExplorerService.getConnectionCredentials(sessionId);
+    public async removeNode(node: ConnectionNode): Promise<void> {
+        const response = await vscode.window.showInformationMessage(
+            LocalizedConstants.ObjectExplorer.NodeDeletionConfirmation(node.label as string),
+            {
+                modal: true,
+            },
+            LocalizedConstants.ObjectExplorer.NodeDeletionConfirmationYes,
+        );
+        if (response === LocalizedConstants.ObjectExplorer.NodeDeletionConfirmationYes) {
+            await this._objectExplorerService.removeNode(node);
+            this.refresh(undefined);
+            await this._connectionManager.connectionStore.removeProfile(
```

Comment:
What's the thinking being saving a reference to connectionManager here instead of keeping this thin and leaving business logic in OEService?